### PR TITLE
Audit usage of memcpy in PAL for Debug

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

**Comment by wtgodbe** [comment]
@dotnet-bot test this please

**Comment by wtgodbe** [comment]
@dotnet-bot retest Ubuntu x64 Release Build and Test

**Comment by wtgodbe** [comment]
@dotnet-bot retest Ubuntu x64 Release Build and Test

**Comment by wtgodbe** [comment]
@dotnet-bot test CI please

**Comment by wtgodbe** [comment]
@dotnet-bot test CI please

File: src/pal/src/cruntime/misc.cpp [hunk 1]
```diff
@@ -634,3 +637,33 @@ void MiscUnsetenv(const char *name)
...
+    {
+        #define memcpy_dummy memcpy
+        #undef memcpy
+        ret = memcpy(dest, src, count);
+        #define memcpy memcpy_dummy
+    }
+    //Otherwise, use memmove since it's overlap-safe 
+    else 
+    {
+        ret = memmove(dest, src, count);
```
Comment (sergiy-k) [inline src/pal/src/cruntime/misc.cpp]: We should not fallback to memmove here. It should be an assert instead. The goal is to find places in the codebase that are using memcpy incorrectly and fix them.

**Comment by wtgodbe** [comment]
@sergiy-k this appears to be working for me locally, if it looks good to you I'll merge after the tests pass

**Comment by wtgodbe** [comment]
@dotnet-bot retest OSX x64 Release Build and Test

**Comment by sergiy-k** [comment]
LGTM

**Pull request merged**